core/templates/domain/exploration/read-only-exploration-backend-api.service.ts

```diff
@@ -76,12 +82,28 @@ export class ReadOnlyExplorationBackendApiService {
       const explorationDataUrl = this._getExplorationUrl(
         explorationId, version);
 
-      this.http.get<FetchExplorationBackendResponse>(
-        explorationDataUrl).toPromise().then(response => {
-        resolve(response);
-      }, errorResponse => {
-        reject(errorResponse.error.error);
-      });
+      // If version is not null, then check whether the data is already cached.
+      // If so, then resolve the Promise with the cached data.
+      if (
+        version && this.versionedExplorationCachingService.isCached(
+          explorationId, version
+        )) {
+        resolve(
+          this.versionedExplorationCachingService
+            .retrieveCachedVersionedExplorationData(explorationId, version));
+      } else {
+        this.http.get<FetchExplorationBackendResponse>(
+          explorationDataUrl).toPromise().then(response => {
+          // If version if not null, then cache the fetched data.
```

Comment:
if ->  is

Reply:
Ditto: be specific about what data refers to

Reply:
Done.